swarm/storage/localstore.go

```diff
@@ -184,3 +184,40 @@ func (ls *LocalStore) Iterator(from uint64, to uint64, po uint8, f func(Address,
 func (ls *LocalStore) Close() {
 	ls.DbStore.Close()
 }
+
+func (ls *LocalStore) Migrate() error {
+	schema, err := ls.DbStore.GetSchema()
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	log.Debug("found schema", "schema", schema, "runtime-schema", CurrentDbSchema)
+	if schema != CurrentDbSchema {
+		// run migrations
+
+		if schema == "" {
+			log.Debug("running migrations for", "schema", schema, "runtime-schema", CurrentDbSchema)
+
+			cleanupFunc := func(c *chunk) bool {
+				valid := false
```

Comment:
don't need the var

Reply:
What do you mean? I think we need it, and this is the same logic as where we run these validators in the LocalStore.

Note that the word Validator is actually confusing here - only one validator has to pass for a chunk to be considered valid, not all of them.

Reply:
Ah sorry yes of course you continue execution after valid. Never mind.